drivers/storage/digitalocean/executor/digitalocean_executor.go

```diff
@@ -0,0 +1,124 @@
+// +build !libstorage_storage_executor libstorage_storage_executor_digitalocean
+
+package executor
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	gofig "github.com/akutz/gofig/types"
+	"github.com/codedellemc/libstorage/api/registry"
+	"github.com/codedellemc/libstorage/api/types"
+	"github.com/codedellemc/libstorage/drivers/storage/digitalocean"
+	doUtils "github.com/codedellemc/libstorage/drivers/storage/digitalocean/utils"
+)
+
+var (
+	diskPrefix = regexp.MustCompile(fmt.Sprintf("^%s(.*)", digitalocean.VolumePrefix))
+	diskSuffix = regexp.MustCompile("part-.*$")
+)
+
+type driver struct {
+	config gofig.Config
+}
+
+func init() {
+	registry.RegisterStorageExecutor(digitalocean.Name, newDriver)
+}
+
+func newDriver() types.StorageExecutor {
+	return &driver{}
+}
+
+func (d *driver) Name() string {
+	return digitalocean.Name
+}
+
+func (d *driver) Init(ctx types.Context, config gofig.Config) error {
+	d.config = config
+	return nil
+}
+
+func (d *driver) InstanceID(ctx types.Context, opts types.Store) (*types.InstanceID, error) {
+	return doUtils.InstanceID(ctx)
+}
+
+func (d *driver) NextDevice(ctx types.Context, opts types.Store) (string, error) {
+	return "", types.ErrNotImplemented
+}
+
+func (d *driver) LocalDevices(ctx types.Context, opts *types.LocalDevicesOpts) (*types.LocalDevices, error) {
+	deviceMap := map[string]string{}
+	diskIDPath := "/dev/disk/by-id"
+
+	dir, _ := ioutil.ReadDir(diskIDPath)
+	for _, device := range dir {
+		switch {
+		case !diskPrefix.MatchString(device.Name()):
+			continue
+		case diskSuffix.MatchString(device.Name()):
+			continue
+		case diskPrefix.MatchString(device.Name()):
+			volumeName := diskPrefix.FindStringSubmatch(device.Name())[1]
+			devPath, err := filepath.EvalSymlinks(
+				fmt.Sprintf("%s/%s", diskIDPath, device.Name()))
+			if err != nil {
+				return nil, err
+			}
+			deviceMap[volumeName] = devPath
+		}
+	}
+
+	ld := &types.LocalDevices{Driver: d.Name()}
+	if len(deviceMap) > 0 {
+		ld.DeviceMap = deviceMap
+	}
+
+	return ld, nil
+}
+
+func (d *driver) Supported(ctx types.Context, opts types.Store) (bool, error) {
+	iid, err := d.InstanceID(ctx, opts)
+	if err != nil {
+		return false, err
+	}
+
+	token := d.token()
+	region := iid.Fields[digitalocean.InstanceIDFieldRegion]
+	client, err := doUtils.Client(token)
+	if err != nil {
+		return false, err
+	}
+
+	regions, _, err := client.Regions.List(nil)
+	if err != nil {
+		return false, err
+	}
+
+	for _, reg := range regions {
+		if reg.Slug == region {
+			for _, feature := range reg.Features {
+				if feature == "storage" {
+					return true, nil
+				}
+			}
+		}
+	}
+
+	return false, nil
+}
+
+func (d *driver) token() string {
+	var token string
+	envToken := os.Getenv("DIGITALOCEAN_ACCESS_TOKEN")
```

Comment:
Hi @protochron,

I'm not going to make this comment on each of the below calls, but this remark does apply to all of the code below related to retrieving configuration properties.

There is no need to check the environment variables directly. Please see this [example](https://github.com/codedellemc/libstorage/blob/master/drivers/storage/ebs/storage/ebs_storage.go#L1321-L1332) for the EBS driver (ignore the backwards-compatability checks). The [gofig](https://github.com/akutz/gofig) package uses @spf13's [Viper](https://github.com/spf13/viper), which in turn enables the assignment of environment variables to defined configuration properties. 

Let's look at [line \#32](https://github.com/protochron/libstorage/blob/f7ce053cd9fec2638ca55cefa79abacfffd83495/drivers/storage/digitalocean/digitalocean.go#L32) in your `digitalocean.go`:

```go
r.Key(gofig.String, "", "", "", Name+"."+"token")
```

In addition to notifying Gofig about the property `digitalocean.token`, the above function call instructs Viper to bind the property `digitalocean.token` to the CLI flag `--digitaloceanToken` and the environment variable `DIGITALOCEAN_TOKEN`. You can actually use Gofig to overwrite which CLI flag or environment variable the property is bound via the [`Key` function's trailing variadic parameter](https://github.com/akutz/gofig/blob/master/gofig_reg.go#L56-L114). 

For example, if we wanted to still use the config property `digitalocean.token` but change the CLI flag to `digiToken` and the environment variable to `DIGI_TOKEN` we could rewrite the above line like so:

```go
r.Key(gofig.String, "", "", "", 
	Name+"."+"token",
	"digiToken",
	"DIGI_TOKEN")
```

Once the configuration properties are bound to a CLI flag or environment variable, invoking `config.GetString("digitalocean.token")` checks first the environment variable, CLI flag argument, and finally a configuration property in that order.